src/main/java/microsoft/exchange/webservices/data/autodiscover/AutodiscoverService.java

```diff
@@ -1511,6 +1512,9 @@ private boolean tryGetEnabledEndpointsForHost(String host,
       HttpWebRequest request = null;
       try {
         request = new HttpClientWebRequest(httpClient, httpContext);
+        if(getWebProxy() != null ){
```

Comment:
thought I yanked it out. pls see the diff from my last commit. May be squashing 4 commits to 1 might summerize better. Let me try that out!
https://github.com/pheenomenon/ews-java-api/commit/ad0b36256ed55956c599a194ca10cc96ca1eaa8e